File: C++/ReverseList.cpp
#include<iostream>

struct ListNode {
   int val;
   struct ListNode *next;
  ListNode(int x) : val(x), next(nullptr) {}
};
class Solution{
    public:
    ListNode* ReverseList(ListNode* head) {
        while (head == nullptr || head->next == nullptr) {
            return head;
        }
        ListNode* prev = nullptr;
        ListNode* next = nullptr;
        while (head != nullptr) {
            next = head->next;
            head->next = prev;
            prev = head;
            head = next;
        }
        return prev;
    }
};

int main(){
    ListNode* head = new ListNode(1);
    head->next=new ListNode(2);
    head->next->next=new ListNode(3);
    Solution solution;
    
    std::cout << "Original list: ";
    for (ListNode* node = head; node != nullptr; node = node->next) {
        std::cout << node->val << " ";
    }
    std::cout << std::endl;
    
    ListNode* reversedHead = solution.ReverseList(head);
    std::cout<<"Reversed list:";
    for(ListNode* node = reversedHead;node!=nullptr;node=node->next){
        std::cout<<node->val<<" ";
    }
    std::cout<<std::endl;

    while(head !=nullptr){
        ListNode*next=head->next;
        delete head;
        head=next;
    }

    return 0;
}

File: python/ReverseList.py
class ListNode:
     def __init__(self, x):
        self.val = x
        self.next = None
        
class Solution:
    def ReverseList(self , head: ListNode) -> ListNode:
        while head ==None:
            return None
        pre =None
        next = None
        while head != None:
            next =head.next
            head.next =pre
            pre = head
            head = next
        return pre

def print_list(node):
    while node:
        print(node.val, end=" -> ")
        node = node.next
    print("None")

head = ListNode(1)
head.next = ListNode(2)
head.next.next = ListNode(3)

print("Original List:")
print_list(head)

solution = Solution()
reversed_head = solution.ReverseList(head)

print("Reversed List:")
print_list(reversed_head)